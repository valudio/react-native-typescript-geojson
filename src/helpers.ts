import { Position } from 'geojson';

import { IOverlay } from './models';

const flatten = (prev: Array<IOverlay>, curr: Array<IOverlay>) =>
  prev.concat(curr);

const makePoint = (c: Position) => ({ latitude: c[1], longitude: c[0] });

const makeLine = (l: any) => l.map(makePoint);

export { flatten, makePoint, makeLine };
