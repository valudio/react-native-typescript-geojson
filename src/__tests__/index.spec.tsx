import React from 'react';
import TestRenderer from 'react-test-renderer';
import { FeatureCollection } from 'geojson';

import Geojson from '..';
import { makePoint } from '../helpers';

test('Geojson with a valid geojson should render the components', () => {
  const alcatraz: FeatureCollection = {
    type: 'FeatureCollection',
    features: [
      {
        type: 'Feature',
        properties: {},
        geometry: {
          type: 'Point',
          coordinates: [-122.42305755615234, 37.82687023785448]
        }
      }
    ]
  };

  const geoJson = TestRenderer.create(<Geojson geojson={alcatraz} />).toJSON();

  expect(geoJson).toMatchSnapshot();
});

test('should put the pin in the correct coordinates', () => {
  const coordinates = [-122.42305755615234, 37.82687023785448];

  const alcatraz: FeatureCollection = {
    type: 'FeatureCollection',
    features: [
      {
        type: 'Feature',
        properties: {},
        geometry: {
          type: 'Point',
          coordinates
        }
      }
    ]
  };

  const geoJson = TestRenderer.create(<Geojson geojson={alcatraz} />).toJSON();

  expect(geoJson).toMatchSnapshot();
  expect(geoJson.props['coordinate']).not.toBeUndefined();
  expect(geoJson.props['coordinate']).toStrictEqual(makePoint(coordinates));
});

test('should change the pin color if prop is provided', () => {
  const alcatraz: FeatureCollection = {
    type: 'FeatureCollection',
    features: [
      {
        type: 'Feature',
        properties: {},
        geometry: {
          type: 'Point',
          coordinates: [-122.42305755615234, 37.82687023785448]
        }
      }
    ]
  };

  const geoJson = TestRenderer.create(
    <Geojson geojson={alcatraz} pinColor="#1ABA85" />
  ).toJSON();

  expect(geoJson).toMatchSnapshot();
  expect(geoJson.props['pinColor']).not.toBeUndefined();
  expect(geoJson.props['pinColor']).toBe('#1ABA85');
});
