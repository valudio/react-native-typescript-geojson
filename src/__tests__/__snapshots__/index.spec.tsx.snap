// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Geojson with a valid geojson should render the components 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "latitude": 37.82687023785448,
      "longitude": -122.42305755615234,
    }
  }
  onPress={[Function]}
  stopPropagation={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      undefined,
    ]
  }
/>
`;

exports[`should change the pin color if prop is provided 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "latitude": 37.82687023785448,
      "longitude": -122.42305755615234,
    }
  }
  onPress={[Function]}
  pinColor="#1ABA85"
  stopPropagation={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      undefined,
    ]
  }
/>
`;

exports[`should put the pin in the correct coordinates 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "latitude": 37.82687023785448,
      "longitude": -122.42305755615234,
    }
  }
  onPress={[Function]}
  stopPropagation={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      undefined,
    ]
  }
/>
`;
